Editor/Window/ManagedEC2Deployment.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Threading.Tasks;
+using AmazonGameLift.Editor;
+using UnityEngine;
+
+namespace AmazonGameLift.Editor
+{
+    public class ManagedEC2Deployment
+    {
+        private readonly DeploymentSettings _model;
+        private readonly FleetParameters _parameters;
+        
+        internal ManagedEC2Deployment(DeploymentSettings model, FleetParameters parameters)
+        {
+            _model = model;
+            _parameters = parameters;
+        }
+        
+        public void UpdateModelFromParameters()
+        {
+            _model.FleetName = _parameters.FleetName;
+            _model.BuildName = _parameters.BuildName;
+            _model.LaunchParameters = _parameters.LaunchParameters;
+            _model.BuildFolderPath = _parameters.GameServerFolder;
+            _model.BuildFilePath = _parameters.GameServerFile;
+            _model.BuildOperatingSystem = _parameters.OperatingSystem;
+        }
+        
+        public void StartDeployment()
+        {
+            UpdateModelFromParameters();
+            _model.GameName = Application.productName.Substring(0, 12);
```

Comment:
What is `Application.productName`? Is there a better way to extract the gameName?
* Is it always a constant?
* Can there be a regex instead if we expect it to match a pattern?

Reply:
`productName` is set within Unity 
![image](https://github.com/aws/amazon-gamelift-plugin-unity/assets/109524503/ee6411fe-9276-4098-b7c5-e25548cc138c)
The substring is because the CloudFormation template has an existing limit on GameName length I am respecting

Reply:
Can someone just go in and change the ProductName? That would then break the substring logic?

Reply:
Yes the produce name is editable, but I'm not sure how likely it is